lib/puppet/provider/image_streamer_resource.rb

```diff
@@ -33,12 +29,16 @@ def self.client
       OneviewSDK::ImageStreamer::Client.new(login_image_streamer)
     end
 
-    def ov_resource_type
+    def client
+      self.class.client
+    end
+
+    def self.ov_resource_type
       api_version = login_image_streamer[:api_version] || 300
       Object.const_get("OneviewSDK::ImageStreamer::API#{api_version}::#{resource_name}")
```

Comment:
We've got some sort of issue when using resource_name on a self method at the moment.
Getting the following error if I try to make the change

```ruby
     Failure/Error: class_name.split('::')[2].split('_').drop(shift_prefix).collect(&:capitalize).join

     NoMethodError:
       undefined method `split' for nil:NilClass
```
Gotta dig somewhat deeper to verify.